### PR TITLE
[Xharness] Fix the test selector.

### LINKED ISSUE #11299: `Xtro (Ignored)` when `src/*` files are modified
### Steps to Reproduce

1. Create a PR with changes in `src`, like https://github.com/xamarin/xamarin-macios/pull/11263

note: this used to work, without using a label, based on the file that were changed inside the PR

### Expected Behavior

Xtro should be run to report binding mistakes

### Actual Behavior

Xtro is ignored

### Environment

`main`

### PR DESCRIPTION
**Pull request by mandel-macaque**
Fix the tests selector by setting the PR number in an env variable that
is not longer linked with Jenkins.

Fixes #11299

### CODE REVIEW

File: tools/devops/automation/templates/build/build.yml [hunk 1]
```diff
@@ -169,6 +169,13 @@ steps:
     if ($buildReason -eq "PullRequest" -or (($buildReason -eq "Manual" -or $buildReason -eq "IndividualCI") -and ($buildSourceBranchName -eq "merge")) ) {
       $configVars.Add("IsPR", "True")
 
+      if ([string]::IsNullOrEmpty($Env:ghprbPullId)) {
+        $configVars.Add("PrID", $Env:ghprbPullId)
+      } else {
+        $configVars.Add("PrID", "")
+      }
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/build/build.yml:176]: We still use if for API diff, I'll remove it in a next PR once we have the API diff and Generator which is running atm.

File: tools/devops/automation/templates/build/build.yml [hunk 2]
```diff
@@ -401,10 +408,12 @@ steps:
   enabled: ${{ parameters.runTests }}
   env:
     IsPR: $(configuration.IsPR)
+    IsPR: $(configuration.PrID)
```
Comment (spouliot) [inline tools/devops/automation/templates/build/build.yml:411]: ^ typo ?

File: tools/devops/automation/templates/build/build.yml (hunk 1 above)
Comment (spouliot) [reply tools/devops/automation/templates/build/build.yml:176]: is that not identical ?

File: tools/devops/automation/templates/build/build.yml (hunk 2 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/build/build.yml:411]: typo.

Comment (mandel-macaque) [inline tools/devops/automation/templates/build/build.yml:411]:
```suggestion
    PrID: $(configuration.PrID)
```

File: tools/devops/automation/templates/build/build.yml (hunk 1 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/build/build.yml:176]: Nope, the yaml is not fwd the env from the previous job. We are using the specific env for the step. No ideal.

File: tools/devops/automation/templates/build/build.yml [hunk 3]
```diff
@@ -169,6 +169,13 @@ steps:
     if ($buildReason -eq "PullRequest" -or (($buildReason -eq "Manual" -or $buildReason -eq "IndividualCI") -and ($buildSourceBranchName -eq "merge")) ) {
       $configVars.Add("IsPR", "True")
 
+      if ([string]::IsNullOrEmpty($Env:ghprbPullId)) {
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/build/build.yml:172]:
```suggestion
      if (-not [string]::IsNullOrEmpty($Env:ghprbPullId)) {
```

File: tools/devops/automation/templates/build/build.yml (hunk 1 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/build/build.yml:176]: good catch, I need to rest

**Review by spouliot — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — COMMENTED** [review]
What's wrong with "ghprbPullId"?

**Comment by mandel-macaque** [comment]
@rolfbjarne bloody VSTS will make an ugly thing GHPRBPULLID

**Comment by rolfbjarne** [comment]
> @rolfbjarne bloody VSTS will make an ugly thing GHPRBPULLID

Yeah, that's ugly. However, `ghprPullId` shows up in a lot of places in our repo:

```
$ git grep ghprbPullId   
Make.config:# Jenkins may set either CHANGE_ID or ghprbPullId
Make.config:else ifneq ($(ghprbPullId),)
Make.config:PULL_REQUEST_ID=$(ghprbPullId)
jenkins/build-api-diff.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/build-api-diff.sh:if test -n "$ghprbPullId" && ./jenkins/fetch-pr-labels.sh --check=skip-public-jenkins; then
jenkins/build.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/build.sh:if test -z "$ghprbPullId"; then
jenkins/build.sh:       echo "Could not find the environment variable ghprbPullId, so forcing a device build."
jenkins/build.sh:       echo "Listing modified files for pull request #$ghprbPullId..."
jenkins/build.sh:       if git diff-tree --no-commit-id --name-only -r "origin/pr/$ghprbPullId/merge^..origin/pr/$ghprbPullId/merge" > .tmp-files; then
jenkins/compare.sh:#    c. Execute 'ghprbPullId=4614 ./compare.sh'
jenkins/compare.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/compare.sh:if test -n "$ghprbPullId"; then
jenkins/compare.sh:if test -z "$ghprbPullId"; then
jenkins/compare.sh:     BASE="origin/pr/$ghprbPullId/merge"
jenkins/fetch-pr-labels.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/fetch-pr-labels.sh:if test -z "$ghprbPullId"; then
jenkins/fetch-pr-labels.sh:     echo "Could not find the environment variable ghprbPullId, so it's not possible to fetch the labels for any pull request."
jenkins/fetch-pr-labels.sh:TMPFILE=".tmp-labels-$ghprbPullId"
jenkins/fetch-pr-labels.sh:     echo "Not downloading labels for pull request #$ghprbPullId because they've already been downloaded."
jenkins/fetch-pr-labels.sh:     echo "Downloading labels for pull request #$ghprbPullId..."
jenkins/fetch-pr-labels.sh:     if ! curl --silent "https://api.github.com/repos/xamarin/xamarin-macios/issues/$ghprbPullId/labels" > "$TMPFILE.dl"; then
jenkins/fetch-pr-labels.sh:             echo "Failed to fetch labels for the pull request $ghprbPullId."
jenkins/fetch-pr-labels.sh:             echo "The pull request $ghprbPullId contains the label $CHECK."
jenkins/fetch-pr-labels.sh:             echo "The pull request $ghprbPullId does not contain the label $CHECK."
jenkins/provision-deps.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/provision-deps.sh:if test -n "$ghprbPullId" && ./jenkins/fetch-pr-labels.sh --check=skip-public-jenkins; then
jenkins/publish-results.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/publish-results.sh:if test -n "$ghprbPullId"; then
jenkins/publish-results.sh:     BRANCH_NAME="pr$ghprbPullId"
jenkins/publish-results.sh:     echo "Neither BRANCH_NAME nor ghprbPullId is set"
jenkins/run-tests.sh:# SC2154: ghprbPullId is referenced but not assigned.
jenkins/run-tests.sh:if test -n "$ghprbPullId" && ./jenkins/fetch-pr-labels.sh --check=skip-public-jenkins; then
tests/xharness/Jenkins/TestSelector.cs:                 if (!int.TryParse (Environment.GetEnvironmentVariable ("ghprbPullId"), out int pullRequest))
tests/xharness/Jenkins/TestSelector.cs:                         MainLog.WriteLine ("The environment variable 'ghprbPullId' was not found, so no pull requests will be checked for test selection.");
tools/devops/automation/templates/build/stage.yml:    ghprbPullId: $[ dependencies.configure.outputs['labels.pr-number'] ]
```

and having multiple variable names for the same concept is confusing and prone to errors. Could we settle on as few as possible?

**Comment by mandel-macaque** [comment]
@rolfbjarne yes, 100% I am going to remove all of them and use a single one.

**Pull request merged**